src/com/sun/jna/Structure.java

```diff
@@ -593,7 +605,7 @@ public void read() {
         }
         finally {
             busy().remove(this);
-            if (reading().get(getPointer()) == this) {
+            if (this instanceof Structure.ByReference && reading().get(getPointer()) == this) {
```

Comment:
Can you tell me why this is necessary?  Does this correctly handle the case where the plain `Structure` class is used but passed as an argument in a method which assumes `ByReference` behavior?

Reply:
The only place where items are added to `reading()` is is some lines above and is using the `this instanceof Structure.ByReference` constrain to add `this` to `reading()`. The result of an `instanceof` check will not change for one object instance in Java, as far as i know. Therefore i think it is safe to use the same constrain removing the items from `reading()` and avoid some lookups in this map.

Reply:
That should have been obvious.  Thanks.